### PR TITLE
feat: Implement product checkout

### PR DESCRIPTION
**Pull request by TPriince**
1. Created checkout workflow - to collect user details (address details included)
2. Implemented editing of user details in the order confirmation segment
3. Integrated Paystack

### CODE REVIEW

File: src/app/magic/checkout/[productId]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { notFound } from "next/navigation";
+import Checkout from "@/components/magic/checkout/Checkout";
+import { getAllProducts } from "@/components/magic/product/getAllProducts";
+
+export async function generateMetadata({
+  params,
+}: {
+  params: { productId: string };
+}): Promise<Metadata> {
+  const res = await getAllProducts(params.productId);
```
Comment (blissuche90) [inline src/app/magic/checkout/[productId]/page.tsx:11]: All this looks like Js, no strong typing.

File: src/components/magic/checkout/sendPayment.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  platform_fee: number;
+  shipping_fee: number;
+  total_paid: number;
+};
+
+export const sendPurchaseDetails = async (data: PurchaseDetailsType) => {
+  const url = `${process.env.NEXT_PUBLIC_API_BASE_URL}order/create`;
+
+  try {
+    const res = await fetch(url, {
```
Comment (blissuche90) [inline src/components/magic/checkout/sendPayment.ts:15]: I think we should have a central function for sending out request that should be used across the entire project.

**Review by blissuche90 — CHANGES_REQUESTED** [review]
Please fix comments

**Comment by TPriince** [comment]
They have all been attended to. I will close this PR and make a new one with the updates. @blissuche90

**Pull request closed without merging**